Simplenote/InterlinkProcessor.swift

```diff
@@ -1,10 +1,32 @@
 import Foundation
 
 
+// MARK: - InterlinkProcessorDelegate
+//
+protocol InterlinkProcessorDelegate: NSObjectProtocol {
+
+    /// Invoked whenever an Autocomplete Row has been selected: The handler should insert the specified text at a given range
+    ///
+    func interlinkProcessor(_ processor: InterlinkProcessor, insert text: String, in range: Range<String.Index>)
+}
+
+
 // MARK: - InterlinkProcessor
 //
 class InterlinkProcessor: NSObject {
 
+    /// Main MOC
+    ///
+    private let viewContext: NSManagedObjectContext
+
+    /// Hosting TextView
+    ///
+    private let parentTextView: SPTextView
+
+    /// Storage Lookup
+    ///
+    private lazy var resultsController = InterlinkResultsController(viewContext: viewContext)
```

Comment:
Do you think we can create `resultsController` in the init and not store `let viewContext: NSManagedObjectContext ` as I think we don't need it for anything else 🤔